XLS-0033d (MPT): Clarify MPTAmount ranges (#244)
XRPLF/XRPL-Standards: XLS-0033d-multi-purpose-tokens/README.md

###### 2.1.1.2.5. `MaximumAmount`

An unsigned 64-bit number that specifies the maximum number of MPTs that can be distributed to non-issuing

> accounts (i.e., `minted`). The maximum value of this field is `0xFFFF'FFFF'FFFF'FFFF`.

Reviewer: ```suggestion
accounts (i.e., `minted`). The default and maximum value is 0x7FFF'FFFF'FFFF'FFFF. However, this may be subject to change in the future, so it's recommended to specify a custom value if a specific limit is needed.
```

something along these lines

###### 2.1.1.2.6. `OutstandingAmount`

An unsigned 64-bit number that specifies the sum of all token amounts that have been minted to all token holders. This
value can be stored on ledger as a `default` type so that when its value is 0, it takes up less space on ledger. This
value is increased whenever an issuer pays MPTs to a non-issuer account, and decreased whenever a non-issuer pays MPTs

> into the issuing account. The maximum value of this field is `0xFFFF'FFFF'FFFF'FFFF`.

Reviewer: ```suggestion
into the issuing account. The maximum value of this field is `0x7FFF'FFFF'FFFF'FFFF`.
```

`2^64 -1` (inclusive). An exception to this rule occurs when MPT amounts are represented in a payment or any other
transaction that requires an `STAmount`. In these instances, the MPT amount values are limited to values between `0` and
`2^63 -1` (inclusive), because this is the range supported by `STAmount`.

As part of the design of MPTs, it was decided to limit the range of MPTs in any `STAmount` so that MPT amounts would
work correctly with the `Number` class, which is used internally in rippled for any many mathematical operations such as
inside the DEX, AMM, or otherwise.

At present, the `Number` class is unable to represent numeric values larger than `2^63 - 307` without loss of precision
or overflow (depending on how the `Number` class's overflow behavior is configured for any particular runtime
operation). While refactoring the `Number` class to support larger internal types was considered, it would have resulted
in performance penalties for all `Number` usages, so we elected to simply limit the size of MPT amounts inside of an
`STAmount` instead. The only real side effect of this design choice is that in certain scenarios where an MPT issuance's

> `MaxAmount` exceeds the value allowed by `STAMount` (i.e., exceeds `2^63-1`), then a single payment, DEX offer, or AMM

Reviewer: ```suggestion
`MaxAmount` exceeds the value allowed by `STAmount` (i.e., exceeds `2^63-1`), then a single payment, DEX offer, or AMM
```

Author: Nice catch!

#### A.1.12. What numeric range do MPT amounts support?

In general, `amount` fields in MPT-related ledger objects are represented using unsigned 64-bit integers that can support values
between `0` and`2^64 -1` (inclusive). An exception to this rule occurs when MPT amounts are represented in a payment or
any other transaction that requires an `STAmount`. In these instances, the MPT amount values are limited to values
between `0` and `2^63 -1` (inclusive), because this is the range supported by `STAmount`.

Because of this limitation, the maximum amounts allowed by any balance tracking field in `MPTokenIssuance` and `MPToken` 
are also limited to values between `0` and `2^63 -1` (inclusive), although these values may be increased in the future 
to enable more tokens to exist. 

For context, MPT amounts in `STAmount` objects are limited to the above range so that MPT amounts will eventually work

> correctly with the `Number` class, which is used internally in rippled for any many mathematical operations such as

Reviewer 2: Typo - "any many".

Author: Good catch. Fixed via b401ae24fd11deab1060f6c392dc72a6fb9fdded
